src/daemon/emitter.cc

```diff
@@ -430,15 +447,17 @@ void Emitter::DoRemoteExecute(const base::WorkerPool& pool, ResolveFn resolver,
 
     // Check that we have a compiler of a requested version.
     net::proto::Status status;
-    if (!SetupCompiler(incoming->mutable_flags(), &status)) {
+    const bool has_compiler_path =
```

Comment:
We may come here two times(for initial shard and once again if shard was down). So to prevent "setting up compiler" twice, we check if it was already setted by checking flags of incoming task.